plugins/scanners/fossid/src/main/kotlin/FossId.kt

```diff
@@ -671,6 +628,34 @@ class FossId internal constructor(
             }
     }
 
+    private suspend fun createIgnoreRules(
+        scanCode: String,
+        excludes: Excludes?,
+        existingRules: List<IgnoreRule> = emptyList(),
+        issues : MutableList<Issue>
+    ) {
+        val excludesRules = excludes?.let {
+            convertRules(it, issues).also {
+                logger.info { "${it.size} rules from ORT excludes have been found." }
+            }
+        }.orEmpty()
+
+        // Create an issue for each legacy rule.
+        val legacyRules = excludesRules.filterLegacyRules(existingRules, issues)
+        if (legacyRules.isNotEmpty()) {
+            logger.warn { "${legacyRules.size} legacy rules have been found." }
+        }
```

Comment:
Could you briefly describe in the commit message how you are ensuring that this code is only executed for the originally third code block? It seems that the "trick" is that `existingRules` is left empty in the other cases, which results in empty `legacyRules`.

I guess what's confusing me here is the way `filterLegacyRules()` works. from its name, I would have guessed that it returns a result based on its receiver `excludesRules`, i.e. that it filters `excludesRules`, whereas in fact it filters the `existingRules` argument.

Reply:
Ok, this more or less became obsolete with [this later change of yours](https://github.com/oss-review-toolkit/ort/pull/8778#discussion_r1648461079)!